src/main.cpp

```diff
@@ -241,14 +235,17 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     mlog("Set total_runtime");
     res.Set("totalRuntime", Napi::Number::New(env, result.total_runtime));
   }
-  if (code != DDWAF_GOOD) {
+  if (code != DDWAF_OK) {
     res.Set("data", Napi::String::New(env, result.data));
   }
-  if (code == DDWAF_MONITOR) {
-    res.Set("action", Napi::String::New(env, "monitor"));
-  }
-  if (code == DDWAF_BLOCK) {
-    res.Set("action", Napi::String::New(env, "block"));
+  if (code == DDWAF_MATCH) {
+    res.Set("status", Napi::String::New(env, "match"));
+    Napi::Array actions = Napi::Array::New(env);
```

Comment:
You should be able to specify the length of the array when creating it:

```
static Napi::Array Napi::Array::New(napi_env env, size_t length);
```

Reply:
Good point, updateing!